launch_testing/launch_testing_examples/launch_testing_examples/check_multiple_node_launch_test.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import random
+import string
+import time
+import unittest
+
+import launch
+import launch.actions
+import launch_ros.actions
+import launch_testing.actions
+import launch_testing.markers
+import pytest
+import rclpy
+from rclpy.node import Node
+
+
+@pytest.mark.launch_test
+@launch_testing.markers.keep_alive
+def generate_test_description():
+    # Default number of nodes to be launched
+    n = 2
+    node_list = []
+    node_names = []
+
+    for i in range(n):
+        node_list.append(
+            launch_ros.actions.Node(
+                executable='talker',
+                package='demo_nodes_cpp',
+                name='demo_node_' + str(i)
+            )
+        )
+        node_names.append('demo_node_' + str(i))
+
+    node_list.append(launch_testing.actions.ReadyToTest())
+    return launch.LaunchDescription(node_list), {'node_list': node_names}
```

Comment:
> We should have something in launch to make it easier, let me see if I can provide an snippet showing how to do it.

In this case it doesn't make much sense do, using either command line arguments or launch arguments to configure tests aren't good ideas (tests shouldn't required any user input).

The way to do it is:

```python3
    node_list = []
    node_names = []
    def generate_nodes(launch_context):
        try:
            n = int(launch_context.launch_configurations['n'])
        except KeyError:
            print("Expected launch argument 'n' to be provided")
        except ValueError:
            print("Expected launch argument 'n' to be an integer")
        for i in range(n):
            node_list.append(
                launch_ros.actions.Node(
                    executable='talker',
                    package='demo_nodes_cpp',
                    name='demo_node_' + str(i)
                )
            )
            node_names.append('demo_node_' + str(i))
        # an opaque function can return a list of action, that will later be executed
        return node_list

    return launch.LaunchDescription([
        launch.actions.DeclareLaunchArgument(name='n', default_value='2'),
        launch.actions.OpaqueFunction(generate_nodes),
        launch_testing.actions.ReadyToTest(),
    ]), {'node_list': node_names}
```